datafusion/core/src/physical_optimizer/dist_enforcement.rs

```diff
@@ -1136,8 +1115,10 @@ mod tests {
             config.execution.target_partitions = 10;
 
             // run optimizer
-            let optimizer = BasicEnforcement {};
+            let optimizer = EnforceDistribution {};
             let optimized = optimizer.optimize($PLAN, &config)?;
+            let optimizer = EnforceSorting {};
```

Comment:
it is slightly confusing that the tests in `EnforceDistribution` also rely on `EnforceSorting` though I understand the reason for this given they both started in the same pass

Maybe we can add a comment like:

```
// These tests also ensure  `EnforceSorting` because they were written prior to the
// separation of `EnforceSorting` and `EnfoceDistribution`
```

Or something like that to give future readers a clue about the rationale for this seemingly strange choice

Reply:
I agree -- just sent a commit with a `NOTE` and a `TODO`.